### PR TITLE
chore: refactored payment to payment-stripe

### PR DESCRIPTION
**Pull request by vackca**

### CODE REVIEW

File: .run/payment.run.xml
Comment (MatthewPattell) [inline .run/payment.run.xml:1]: rename run config to payment-stripe

File: configs/config.local.json [hunk 1]
```diff
@@ -110,14 +110,14 @@
     "microservice": "payment",
```
Comment (MatthewPattell) [inline configs/config.local.json]: ?

File: configs/config.local.json [hunk 2]
```diff
@@ -110,14 +110,14 @@
...
-      "paymentOptions": {
-        "methods": ["bancontact", "card"],
-        "apiKey": "your test key from stripe or other service",
-        "config": {
+        "paymentMethods": ["bancontact", "card"],
+        "paymentApiKey": "your test key from stripe or other service",
+        "paymentConfig": {
           "apiVersion": "2022-11-15"
-        }
       },
```
Comment (MatthewPattell) [inline configs/config.local.json]: we don't need prefix `payment`

File: configs/config.local.json [hunk 3]
```diff
@@ -110,14 +110,14 @@
...
-      "paymentProvider": "stripe"
+      "payoutCoeff": 0.3,
+      "webhookKey": "your test webhook key from stripe or other service",
+      "paymentFees": { "stableUnit": 30, "paymentPercent": 2.9 }
```
Comment (MatthewPattell) [inline configs/config.local.json]: the same

File: microservices/payment-stripe/README.md [hunk 4]
```diff
@@ -11,6 +11,12 @@ This microservice provides payments mechanism.
 - [MEMORY USAGE](#memory-usage)
 
 ### <a id="environments"></a>ENVIRONMENTS:
+- `PAYMENT_API_KEY` - Stripe api key to connect ms with stripe sdk. Default: `example`
+- `PAYMENT_CONFIG` - JSON onject which contains apiVersion property required by Stripe. Default: `'{"apiVersion": "2022-11-15"}'`
+- `PAYMENT_METHODS` - Payment methods allowed using in Stripe. Default: `'["bancontact", "card"]'`
+- `WEBHOOK_KEY` - Stripe webhook key to connect ms and work with stripe webhook service. Example: `whsec_c6332a6b339173127d2e6c813112e2f2323b4b2b10eea5ac17c5649a60cf335e`
+- `PAYOUT_COEFF` - Number for calculating amount of money for transferring to the product owners. Default: `0.3`
+- `PAYMENT_FEES` - Fees that takes while creating payment intent. Default: `'{ "stableUnit": 30, "paymentPercent": 2.9 }'`
```
Comment (MatthewPattell) [inline microservices/payment-stripe/README.md]: the same

File: microservices/payment-stripe/package.json [hunk 5]
```diff
@@ -28,18 +28,18 @@
   "author": "Mikhail Yarmaliuk",
   "license": "Apache-2.0",
   "scripts": {
-    "test": "cross-env NODE_ENV=tests TS_NODE_COMPILER_OPTIONS={\\\"target\\\":\\\"es6\\\"} mocha --harmony --no-warnings  __tests__ --require ts-node/register --recursive --unhandled-rejections=strict --exit",
-    "lint:check": "eslint --ext \".ts,.tsx\" \"src/**/*.{ts,tsx,*.ts,*tsx}\"",
-    "lint:format": "eslint --fix --ext \".ts,.tsx\" \"src/**/*.{ts,tsx,*.ts,*tsx}\"",
-    "ts:check": "tsc --project ./tsconfig.json --skipLibCheck --noemit",
+    "test": "cross-env NODE_ENV=tests TS_NODE_COMPILER_OPTIONS={\\\\\\\\\\\\\\\"target\\\\\\\\\\\\\\\":\\\\\\\\\\\\\\\"es6\\\\\\\\\\\\\\\"} mocha --harmony --no-warnings  __tests__ --require ts-node/register --recursive --unhandled-rejections=strict --exit",
+    "lint:check": "eslint --ext \\\\\\\".ts,.tsx\\\\\\\" \\\\\\\"src/**/*.{ts,tsx,*.ts,*tsx}\\\\\\\"",
+    "lint:format": "eslint --fix --ext \\\\\\\".ts,.tsx\\\\\\\" \\\\\\\"src/**/*.{ts,tsx,*.ts,*tsx}\\\\\\\"",
```
Comment (MatthewPattell) [inline microservices/payment-stripe/package.json]: wtf?

File: microservices/payment-stripe/package.json [hunk 6]
```diff
@@ -28,18 +28,18 @@
...
+    "ts:check": "tsc --project tsconfig.json --skipLibCheck --noemit",
```
Comment (MatthewPattell) [inline microservices/payment-stripe/package.json]: ?

File: microservices/payment-stripe/src/config/remote.ts [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
 const defaultConfig: IRemoteConfig = {
-  paymentProvider: CONST.PAYMENT_PROVIDER as PaymentProvider,
-  paymentOptions: JSON.parse(CONST.PAYMENT_OPTIONS),
-  paymentWebhookKey: CONST.PAYMENT_WEBHOOK_KEY,
+  paymentMethods: CONST.PAYMENT_METHODS,
+  paymentApiKey: CONST.PAYMENT_API_KEY,
+  paymentConfig: CONST.PAYMENT_CONFIG,
+  webhookKey: CONST.WEBHOOK_KEY,
   payoutCoeff: CONST.PAYOUT_COEFF,
   paymentFees: JSON.parse(CONST.PAYMENT_FEES),
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/config/remote.ts]: the same

File: microservices/payment-stripe/src/constants/index.ts [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
   ...GetConstants({ msNameDefault, version, isBuild, packageName: name, withDb: true }),
-  PAYMENT_PROVIDER: process.env.PAYMENT_PROVIDER ?? PaymentProvider.STRIPE,
-  PAYMENT_OPTIONS: process.env.PAYMENT_OPTIONS ?? '{}',
-  PAYMENT_WEBHOOK_KEY: process.env.PAYMENT_WEBHOOK_KEY ?? '',
+  PAYMENT_API_KEY: process.env.PAYMENT_API_KEY ?? '',
+  PAYMENT_CONFIG: JSON.parse(process.env.PAYMENT_CONFIG ?? '{"apiVersion": "2022-11-15"}'),
+  PAYMENT_METHODS: JSON.parse(process.env.PAYMENT_METHODS ?? '["bancontact", "card"]'),
+  WEBHOOK_KEY: process.env.STRIPE_WEBHOOK_KEY ?? '',
   PAYOUT_COEFF: Number(process.env.PAYOUT_COEFF) ?? 0.3,
   PAYMENT_FEES: process.env.PAYMENT_FEES ?? '{ "stableUnit": 30, "paymentPercent": 2.9 }',
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/constants/index.ts]: the same

File: microservices/payment-stripe/src/interfaces/remote-config.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * Microservice remote config
+ */
+export interface IRemoteConfig {
+  webhookKey?: string;
+  payoutCoeff?: number;
+  paymentFees?: IFees;
+  paymentMethods?: string[];
+  paymentApiKey?: string;
+  paymentConfig?: Stripe.StripeConfig;
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/interfaces/remote-config.ts]: the same

File: microservices/payment-stripe/src/services/payment-gateway/stripe.ts [hunk 10]
```diff
@@ -102,19 +100,21 @@ class Stripe extends Abstract {
...
+    apiKey: string,
+    stripeConfig: StripeTypes.StripeConfig,
+    methods: string[],
   ) {
-    super(paymentProvider, paymentOptions, manager);
+    super(manager);
 
-    this.paymentEntity = new StripeSdk(paymentOptions.apiKey, paymentOptions.config);
+    this.methods = methods;
+    this.paymentEntity = new StripeSdk(apiKey, stripeConfig);
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/services/payment-gateway/stripe.ts]: move to abstract class

Comment (MatthewPattell) [inline microservices/payment-stripe/src/services/payment-gateway/stripe.ts]:
```suggestion
    this.methods = methods;
    this.sdk = new StripeSdk(apiKey, stripeConfig);
```

File: microservices/payment-stripe/src/services/payment-gateway/stripe.ts [hunk 11]
```diff
@@ -102,19 +100,21 @@ class Stripe extends Abstract {
   /**
    * @protected
    */
-  protected readonly paymentOptions: IStripeOptions;
+  protected readonly methods: string[];
 
   /**
    * @constructor
    */
   constructor(
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/services/payment-gateway/stripe.ts]: you don't need constructor here

File: microservices/payment-stripe/src/services/payment-gateway/factory.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+ */
+class Factory {
+  /**
+   * Create gateway instance
+   */
+  public static async create(manager: EntityManager): Promise<Stripe> {
+    const { paymentConfig, paymentMethods, paymentApiKey } = await remoteConfig();
+
+    if (!paymentConfig || !paymentApiKey || !paymentMethods) {
+      throw new Error(`Payment options or api key or payment methods for stripe are not provided`);
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/services/payment-gateway/factory.ts]:
```suggestion
      throw new Error('Payment options or api key or payment methods for stripe are not provided');
```

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/payment-stripe/package.json (hunk 5 above)
Comment (vackca) [reply microservices/payment-stripe/package.json]: IDE bug while moving files

File: microservices/payment-stripe/package.json (hunk 6 above)
Comment (vackca) [reply microservices/payment-stripe/package.json]: IDE bug while moving files

File: microservices/payment-stripe/src/constants/index.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+const msNameDefault = 'payment-stripe';
+
+const constants = {
+  ...GetConstants({ msNameDefault, version, isBuild, packageName: name, withDb: true }),
+  API_KEY: process.env.API_KEY ?? '',
+  CONFIG: JSON.parse(process.env.CONFIG ?? '{"apiVersion": "2022-11-15"}'),
+  PAYMENT_METHODS: JSON.parse(process.env.PAYMENT_METHODS ?? '["bancontact", "card"]'),
+  WEBHOOK_KEY: process.env.STRIPE_WEBHOOK_KEY ?? '',
+  PAYOUT_COEFF: Number(process.env.PAYOUT_COEFF) ?? 0.3,
+  FEES: process.env.FEES ?? '{ "stableUnit": 30, "paymentPercent": 2.9 }',
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/constants/index.ts]: for constant we should add `MS_` prefix

Comment (MatthewPattell) [reply microservices/payment-stripe/src/constants/index.ts]: only here

File: microservices/payment-stripe/src/services/payment-gateway/factory.ts [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+ * Payment gateway factory
+ */
+class Factory {
+  /**
+   * Create gateway instance
+   */
+  public static async create(manager: EntityManager): Promise<Stripe> {
+    const { config, paymentMethods, apiKey } = await remoteConfig();
+
+    console.log({ config, paymentMethods, apiKey }, { config, paymentMethods, apiKey });
```
Comment (MatthewPattell) [inline microservices/payment-stripe/src/services/payment-gateway/factory.ts]: ???

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/payment-stripe/src/constants/index.ts (hunk 13 above)
Comment (MatthewPattell) [reply microservices/payment-stripe/src/constants/index.ts]: also update readme.md

File: .run/payment-stripe.run.xml [hunk 15]
```diff
@@ -1,5 +1,5 @@
 <component name="ProjectRunConfigurationManager">
-  <configuration default="false" name="payment" type="NodeJSConfigurationType" folderName="microservices" node-parameters="--require ts-node/register" path-to-js-file="src/index.ts" working-dir="$PROJECT_DIR$/microservices/payment">
+  <configuration default="false" name="payment" type="NodeJSConfigurationType" folderName="microservices" node-parameters="--require ts-node/register" path-to-js-file="$PROJECT_DIR$/microservices/payment-stripe/src/index.ts" working-dir="$PROJECT_DIR$/microservices/payment-stripe">
     <envs>
       <env name="DB_FROM_CONFIG_MS" value="0" />
       <env name="PAYMENT_WEBHOOK_KEY" value="" />
```
Comment (MatthewPattell) [inline .run/payment-stripe.run.xml]:
```suggestion
      <env name="MS_WEBHOOK_KEY" value="" />
```

**Review by MatthewPattell — APPROVED** [review]

**Pull request merged**